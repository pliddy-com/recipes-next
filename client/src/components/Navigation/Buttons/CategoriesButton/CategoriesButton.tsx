import MenuIcon from '@mui/icons-material/Menu';

import NavIconButton from '../NavIconButton/NavIconButton';

interface CategoriesButtonProps {
  onClick: VoidFunction;
  hideLabel?: boolean;
}

const CategoriesButton = ({ onClick, hideLabel }: CategoriesButtonProps) => {
  const ariaLabel = 'open categories menu';
  const className = 'menu-button';
  const label = 'Categories';

  return (
    <NavIconButton
      ariaLabel={ariaLabel}
      className={className}
      hideLabel={hideLabel}
      icon={<MenuIcon />}
      label={label}
      onClick={onClick}
    />
  );
};

export default CategoriesButton;
