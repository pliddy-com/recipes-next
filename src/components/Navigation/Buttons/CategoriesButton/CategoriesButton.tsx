import Button from '@mui/material/Button';
import IconButton from '@mui/material/IconButton';

import MenuIcon from '@mui/icons-material/Menu';

interface MenuButtonProps {
  onClick: VoidFunction;
  hideLabel?: boolean;
}

const MenuButton = ({ onClick, hideLabel = false }: MenuButtonProps) =>
  hideLabel ? (
    <IconButton
      aria-label="open categories menu"
      className="menu"
      color="inherit"
      edge="start"
      onClick={onClick}
      role="button"
    >
      <MenuIcon />
    </IconButton>
  ) : (
    <Button
      aria-label="open categories menu"
      className="menu"
      color="primary"
      onClick={onClick}
      role="button"
      size="large"
      startIcon={<MenuIcon />}
    >
      Categories
    </Button>
  );

export default MenuButton;
