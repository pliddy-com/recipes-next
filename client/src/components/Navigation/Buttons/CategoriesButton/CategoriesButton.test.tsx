import '@testing-library/jest-dom';
import { fireEvent, render } from '@testing-library/react';

import CategoriesButton from './CategoriesButton';

describe('CategoriesButton', () => {
  it('renders a menu button with a label', () => {
    const callback = jest.fn();

    const { queryByRole } = render(<CategoriesButton onClick={callback} />);

    // assert that component has been rendered
    const component = queryByRole('button', { name: 'open categories menu' });

    // assert that callback is called on click
    component && fireEvent.click(component);
    expect(callback).toBeCalled();

    // assert that the component matches the existing snapshot
    expect(component).toMatchSnapshot();
  });

  describe('when the hideLabel flag is true', () => {
    it('renders an icon button', () => {
      const callback = jest.fn();

      const { queryByRole } = render(
        <CategoriesButton onClick={callback} hideLabel={true} />
      );

      // assert that component has been rendered
      const component = queryByRole('button', { name: 'open categories menu' });

      // assert that callback is called on click
      component && fireEvent.click(component);
      expect(callback).toBeCalled();

      // assert that the component matches the existing snapshot
      expect(component).toMatchSnapshot();
    });
  });
});
