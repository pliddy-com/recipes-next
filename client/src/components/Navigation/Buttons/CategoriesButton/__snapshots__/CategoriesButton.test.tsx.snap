// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoriesButton renders a menu button with a label 1`] = `
<button
  aria-label="open categories menu"
  class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeLarge MuiButton-textSizeLarge MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeLarge MuiButton-textSizeLarge menu-button css-kpm0tx-MuiButtonBase-root-MuiButton-root"
  role="button"
  tabindex="0"
  type="button"
>
  <span
    class="MuiButton-startIcon MuiButton-iconSizeLarge css-htszrh-MuiButton-startIcon"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="MenuIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
      />
    </svg>
  </span>
  Categories
  <span
    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
  />
</button>
`;

exports[`CategoriesButton when the hideLabel flag is true renders an icon button 1`] = `
<button
  aria-label="open categories menu"
  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart MuiIconButton-sizeMedium menu-button css-1d6w9lk-MuiButtonBase-root-MuiIconButton-root"
  role="button"
  tabindex="0"
  type="button"
>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
    data-testid="MenuIcon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
    />
  </svg>
  <span
    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
  />
</button>
`;
